### PR TITLE
Fix GUI detection colors by rendering the original image

### LINKED ISSUE #524: [Bug] Inference tab renders normalized tensors without inverse normalization.
The Inference tab in `tabs/inference.py` displays the normalized image tensor directly without applying inverse normalization before rendering:
```python
img_to_draw = transforms.ToPILImage()(sample_tensor[0])
```

This causes the output image to appear with distorted, psychedelic colors instead of the original image, making the inference visualization incorrect and misleading. The image tensor should be de-normalized before display so the output image appears correctly with original colours.

## Proposed Fix
Add a `denormalize` helper function in `tabs/inference.py` and apply it before rendering:
```python
def denormalize(tensor, mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225]):
    mean = torch.tensor(mean).view(3, 1, 1)
    std = torch.tensor(std).view(3, 1, 1)
    return torch.clamp(tensor.cpu() * std + mean, 0, 1)

denormalized = denormalize(sample_tensor[0])
img_to_draw = transforms.ToPILImage()(denormalized)
```

### PR DESCRIPTION
**Pull request by divyanshAgarwal123**
## What changed

The image detection tab was converting the model input tensor back to a display image. When normalization is configured, that tensor no longer contains display-ready RGB values.

This change draws the predictions on the original uploaded RGB image instead. The prediction path already maps bounding boxes back to the original image coordinates. A regression test verifies that the original image is passed to the drawing function and that the normalized sample is not requested.

## Reproduction

With a small RGB sample, the current conversion changed pixel (128, 64, 32) to (18, 23, 195).

## Checks

- Black formatting check
- pytest: 47 passed

Fixes #524

### CODE REVIEW

**Comment by divyanshAgarwal123** [comment]
Hi @dpascualhe, just following up on #631. It is still open and mergeable, and the focused regression test plus the full local suite (47 tests) pass. Please let me know if you’d like any changes or additional coverage. Thanks!
